### PR TITLE
Fixed #19605 Don't static compile disabled modules

### LINKED ISSUE #19605: Don't static compile disabled modules
### Summary (*)

When running static-content:deploy, there are disabled module content in pub/static.

### Examples (*)

Disable modules that have frontend resources.
Run `bin/magento setup:static-content:deploy`
Observe that `pub/static/...` contains the resources.

### Proposed solution

Check the disabled modules and skip them when compiling static content

### PR DESCRIPTION
**Pull request by shikhamis11**
Fixed issue #19605 Don't static compile disabled modules

**Summary** 
 When running static-content:deploy, there are disabled module content in pub/static.

**Examples** 

Disable modules that have frontend resources.
Run bin/magento setup:static-content:deploy
Observe that pub/static/... contains the resources.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by milindsingh** [comment]
@orlangur why "Community Insider: Cedcommerce" label is not added automatically by @magento-engcom-team ?

**Comment by shikhamis11** [comment]
@orlangur may I know why you have added "Progress: on hold" label.

File: app/code/Magento/Deploy/Collector/Collector.php [hunk 1]
```diff
@@ -43,6 +44,13 @@ class Collector implements CollectorInterface
      * @var PackageFactory
      */
     private $packageFactory;
+    
+    /**
+     * Module Manager object
```
Comment (orlangur) [inline app/code/Magento/Deploy/Collector/Collector.php]: Such comment is not really useful, please leave type hint only.

File: app/code/Magento/Deploy/Collector/Collector.php [hunk 2]
```diff
@@ -65,11 +73,13 @@ class Collector implements CollectorInterface
     public function __construct(
         SourcePool $sourcePool,
         FileNameResolver $fileNameResolver,
-        PackageFactory $packageFactory
+        PackageFactory $packageFactory,
+        Manager $moduleManager
```
Comment (orlangur) [inline app/code/Magento/Deploy/Collector/Collector.php]: Please inject new dependencies in backward-compatible manner: https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter

**Review by orlangur — CHANGES_REQUESTED** [review]
@shikhamis11 looks really clean and nice!

Please apply some minor review notes and do commit amend + force push.

`on hold` was to get a small feedback in Slack if there could be some undesired consequences by this change. For example, URL Rewrites are created even for disabled products so that you can enable them later without any collision.

**Comment by shikhamis11** [comment]
@orlangur I have made changes please check

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3701](https://jira.corp.magento.com/browse/ENGCOM-3701) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**